tests/unit/dataactvalidator/test_b21_object_class_program_activity.py

```diff
@@ -0,0 +1,140 @@
+from random import randint
+
+from tests.unit.dataactcore.factories.domain import SF133Factory, TASFactory
+from tests.unit.dataactcore.factories.job import SubmissionFactory
+from tests.unit.dataactcore.factories.staging import ObjectClassProgramActivityFactory
+from tests.unit.dataactvalidator.utils import error_rows, number_of_errors, query_columns
+
+
+_FILE = 'b21_object_class_program_activity'
+
+
+def test_column_headers(database):
+    expected = {
+        'uniqueid_TAS', 'uniqueid_DisasterEmergencyFundCode', 'row_number', 'allocation_transfer_agency',
+        'agency_identifier', 'beginning_period_of_availa', 'ending_period_of_availabil',
+        'availability_type_code', 'main_account_code', 'sub_account_code', 'disaster_emergency_fund_code'
+    }
+    assert expected == set(query_columns(_FILE, database))
+
+
+def test_success_populated_ata(database):
+    """ Tests that TAS/DEFC for SF-133 are present in File B """
+    submission_id = randint(1000, 10000)
+    tas, period, year, code = 'some-tas', 2, 2002, 'some-code'
+
+    sf1 = SF133Factory(tas=tas, period=period, fiscal_year=year, allocation_transfer_agency=code,
+                       agency_identifier='some-other-code', disaster_emergency_fund_code='N')
+    submission = SubmissionFactory(submission_id=submission_id, reporting_fiscal_period=period,
+                                   reporting_fiscal_year=year, cgac_code=code, is_quarter_format=False)
+    op = ObjectClassProgramActivityFactory(tas=tas, disaster_emergency_fund_code='n', submission_id=submission_id)
+
+    assert error_rows(_FILE, database, models=[sf1, op], submission=submission) == []
```

Comment:
Any reason why this and the test below don't use `number_of_errors`?

Reply:
Because I copy-pasted from existing tests and just left it. Did you want it updated?

Reply:
Yeah just for consistency with the other tests.